crash when using 'sun position' shortcut

Since `v0.14.0` the 'sun position' shortcut causes a crash. 

From the launcher or home screen, long press `Suntimes -> Sun Position`

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.forrestguice.suntimeswidget/com.forrestguice.suntimeswidget.SuntimesActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.util.TimeZone com.forrestguice.suntimeswidget.calculator.SuntimesRiseSetDataset.timezone()' on a null object reference
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3635)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3792)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2210)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7839)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.util.TimeZone com.forrestguice.suntimeswidget.calculator.SuntimesRiseSetDataset.timezone()' on a null object reference
        at com.forrestguice.suntimeswidget.LightMapDialog.setData(LightMapDialog.java:132)
        at com.forrestguice.suntimeswidget.SuntimesActivity.showLightMapDialog(SuntimesActivity.java:2102)
        at com.forrestguice.suntimeswidget.SuntimesActivity.handleIntent(SuntimesActivity.java:279)
        at com.forrestguice.suntimeswidget.SuntimesActivity.onCreate(SuntimesActivity.java:258)
        at android.app.Activity.performCreate(Activity.java:8051)
        at android.app.Activity.performCreate(Activity.java:8031)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1329)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3608)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3792) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2210) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loopOnce(Looper.java:201) 
        at android.os.Looper.loop(Looper.java:288) 
        at android.app.ActivityThread.main(ActivityThread.java:7839) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003) 
```
Tested with `v0.14.1` (crashes), and with `v0.13.19` (works correctly).

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,7 @@
 * changes default 'time zone mode' back to `system`; reverts change from 7c288be (#565).
 * adds extras to SuntimesActivity intent; `ACTION_VIEW_SUN` and `ACTION_VIEW_WORLDMAP` now accept `EXTRA_SHOW_DATE`; `ACTION_ADD_ALARM` accepts `EXTRA_SOLAREVENT`.
 * updates translation to Norwegian (nb) (#568 by FTno).
+* updates translations to Polish (pl) and Esperanto (eo) (#571 by Verdulo).
 
 ### v0.14.1 (2022-02-22)
 * fixes crash when using "current location" (#556).

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,13 @@
 ### ~
 
+### v0.14.2 (2022-03-14)
+* fixes crash when using 'sun position' app shortcut (#567).
+* fixes bug where "search places doesn't work" (#566).
+* fixes bug where '1x1 moon widget' illumination is always displayed (fails to be hidden) (#563).
+* changes default 'time zone mode' back to `system`; reverts change from 7c288be (#565).
+* adds extras to SuntimesActivity intent; `ACTION_VIEW_SUN` and `ACTION_VIEW_WORLDMAP` now accept `EXTRA_SHOW_DATE`; `ACTION_ADD_ALARM` accepts `EXTRA_SOLAREVENT`.
+* updates translation to Norwegian (nb) (#568 by FTno).
+
 ### v0.14.1 (2022-02-22)
 * fixes crash when using "current location" (#556).
 * fixes bug "lightmap for tomorrow card fails to display" (#557).

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -23,8 +23,8 @@ android
         minSdkVersion 10
         //noinspection ExpiredTargetSdkVersion,OldTargetApi
         targetSdkVersion 25
-        versionCode 85
-        versionName "0.14.1"
+        versionCode 86
+        versionName "0.14.2"
 
         buildConfigField "String", "GIT_HASH", "\"${getGitHash()}\""
 

--- a/fastlane/metadata/android/en-US/changelogs/86.txt
+++ b/fastlane/metadata/android/en-US/changelogs/86.txt
@@ -1,0 +1,3 @@
+- fixes crash when using 'sun position' app shortcut (#567).
+- fixes bug where "search places doesn't work" (#566).
+- updates translation to Norwegian.

--- a/fastlane/metadata/android/en-US/changelogs/86.txt
+++ b/fastlane/metadata/android/en-US/changelogs/86.txt
@@ -1,3 +1,4 @@
 - fixes crash when using 'sun position' app shortcut (#567).
 - fixes bug where "search places doesn't work" (#566).
 - updates translation to Norwegian.
+- updates translation to Polish and Esperanto.

